syne_tune/optimizer/schedulers/searchers/bore/bore.py

```diff
@@ -293,5 +304,24 @@ def _update(self, trial_id: str, config: dict, result: dict):
         self.inputs.append(self._hp_ranges.to_ndarray(config))
         self.targets.append(result[self._metric])
 
+    def register_pending(
+        self,
+        trial_id: str,
+        config: Optional[dict] = None,
+        milestone: Optional[int] = None,
+    ):
+        if self._allow_duplicates and trial_id not in self._config_for_trial_id:
+            if config is not None:
+                self._config_for_trial_id[trial_id] = config
+            else:
+                logger.warning(
+                    f"register_pending called for trial_id {trial_id} without passing config"
+                )
+
+    def evaluation_failed(self, trial_id: str):
+        if self._allow_duplicates and trial_id in self._config_for_trial_id:
+            # Blacklist this configuration
+            self._excl_list.add(self._config_for_trial_id[trial_id])
```

Comment:
Only get `trial_id` here, but need to know the config. This  is why `_config_for_trial` is needed, only if `allow_duplicates`.